### PR TITLE
Allow websocket URL to contain query string params

### PR DESCRIPTION
**Pull request by jatt**
Refactored WebSocketService#normalizeURL method to handle URL with query params correctly (add / before ?, not to the end of the URL).

### CODE REVIEW

File: app/services/websockets.js [hunk 1]
```diff
@@ -85,11 +85,19 @@ export default Ember.Service.extend({
   normalizeURL(url) {
     var parsedUrl = new URI(url);
 
-    if(parsedUrl.path() === '/' && url.slice(-1) !== '/') {
-      return url + '/';
+    if(parsedUrl.path() !== '/') {
```
Comment (thoov) [inline app/services/websockets.js]: Just to clean this up lets make the logic like this:

```js
normalizeURL(url) {
  var parsedUrl = new URI(url);
  var path = parsedUrl.path();
  var query = parsedUrl.query();

  if(path !== '/' && query === '' && url.slice(-1) !== '/') {
    return url + '/'';
  }
  else if(url.indexOf('/?') === -1) {
    return url.replace('?', '/?');
  }

  return url;
}

**Comment by thoov** [comment]
Thanks @jatt for the PR! Once you address the comment above I will merge

**Comment by thoov** [comment]
We also need to update this: https://github.com/thoov/ember-websockets/blob/master/app/services/websockets.js#L85-L93

I would suggest refactoring the code to use a common function in both places.

**Comment by jatt** [comment]
I refactored this normalizeURL function into a separate mixin that is included into all 3 services. Also I cleaned up a method a bit, but it is now as simple as it can be, your code doesn't cover all cases and tests fail.

**Pull request merged**